boogiestats/boogie_ui/views.py

```diff
@@ -712,3 +714,88 @@ def get_context_data(self, **kwargs):
 
     def get_queryset(self):
         return None
+
+
+class PlayerWrappedView(generic.base.TemplateView):
+    template_name = "boogie_ui/player_wrapped.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        player_id = self.kwargs["player_id"]
+        year = self.kwargs["year"]
+        player = Player.get_or_404(id=player_id)
+        context["player"] = player
+        context["year"] = year
+        scores = player.scores.filter(submission_date__year=year)
+        context["num_scores"] = scores.count()
+        context["num_charts_played"] = scores.filter(is_itg_top=True).count()
+        context["one_star"] = scores.filter(is_itg_top=True, itg_score__gte=9600, itg_score__lt=9800).count()
+        context["two_stars"] = scores.filter(is_itg_top=True, itg_score__gte=9800, itg_score__lt=9900).count()
+        context["three_stars"] = scores.filter(is_itg_top=True, itg_score__gte=9900, itg_score__lt=10000).count()
+        context["four_stars"] = scores.filter(is_itg_top=True, itg_score=10000).count()
+        context["five_stars"] = scores.filter(is_ex_top=True, ex_score=10000).count()
```

Comment:
really gotta refactor that sometime soon because it's repeated like 3 times already

Reply:
created #148 to follow this up